### ISSUE TITLE
Support using along terraform-google-provider 4.x.x

### ISSUE THREAD

**Issue by johnmartel** [terraform-google-modules/terraform-google-gcloud#107]
### TL;DR

The current constraint on the version of `terraform-google-provider` forbids using on projects that have switched to the 4.x series of the provider.

### Additional information

We would like to use this module to invoke `gcloud` to create resources not yet supported by the `google` and `google-beta` providers. These resources have dependencies on other regular resources created with the providers (for instance, regional network-endpoint-group backed by an api-gateway). I blindly tried this module here but the version constraint prevents it: we are already migrated to 4.0.0 of `google` and ` google-beta`.

**Comment by johnmartel** [comment]
This is closed by https://github.com/terraform-google-modules/terraform-google-gcloud/pull/108
